### DISCUSSION TITLE
Modifying Request Body in routerUse

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#5174] [Q&A] [answered]
Hello,

Is it possible to write the response body (`'hello'` in this example) in the router?

```js
routerUse((next) => {
    return (c) => {
        const body = $apis.requestInfo(c).data;
        body.name = 'hello'; // <-- this
        return next(c);
    };
});
```

Thank you

**Comment by ganigeorgiev** [comment, marked as answer]
`$apis.requestInfo(c).data` returns a cached parsed body data that is used usually as part of the API rule checks.

Technically it is possible to modify the request body (or more accurately to replace it) but I don't think this can be done with the JS `pb_hooks` because you'll need to create a new `io.ReadCloser` somehow (new route bindings to the JSVM are not planned for now as the echo router has been already replaced as part of the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355)).

Also note that how the body is parsed and modified/replaced would depend on its content-type (`application/json` or `multipart/form-data`).

In any case, if you need further assistance please don't treat the discussions as llm chat and try to elaborate more on your use case to understand the bigger picture and why this is needed in the first place.

**Reply by AfdhalAtiffTan** [reply]
Hi @ganigeorgiev, 

Forgive me if I came across as rude.

My intention was not to burden you with unnecessary details.

Basically I am working on #5139. 

I am using bash scripts (inotifywait, sha512sum, unzip, zip, etc.) to automatically sign the generated backup then encapsulate it in a .signed.zip.

It will sign and encapsulate it during backup download, then decapsulate and verify the signature during upload.

For this to work, I need to make the backup endpoint to exclude file extension.
